pkgdb/src/resolver/environment.cc

```diff
@@ -82,6 +81,20 @@ Environment::getCombinedRegistryRaw()
                 {
                   input = locked->second;
                 }
+              /* Lock the input if it's not in the lock. */
+              else
+                {
+                  if ( ! store.has_value() )
+                    {
+                      store = NixStoreMixin().getStore();
```

Comment:
quib: this looks weird, but we seem to already use this construct elsewhere

Reply:
Yeah I just copied that from what I saw before:
```
  [[nodiscard]] RegistryRaw
  getLockedRegistry( const nix::ref<nix::Store> & store
                     = NixStoreMixin().getStore() ) const
```

But can change if there's a better way